src/test/java/core/basesyntax/strategy/handlerimpl/PurchaseOperationTest.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.strategy.handlerimpl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+
+public class PurchaseOperationTest {
```

Comment:
1. All existing test methods  from `BalanceOperationTest`, `PurchaseOperationTest` and other operations test classes should be moved to the one class OperationStrategyImplTest. Because now this test classes check only returned class name, not the business logic.
2. The classes `BalanceOperationTest`, `PurchaseOperationTest` and other operations test classes should contains the test of result of operation. E.g. we have `10` bananas in storage, purchase `4`, assert that `6` bananas left at storage. Also add test methods for exceptions that can be thrown. 
Part of test cases already implemented in `FruitShopServiceImplTest` you can move these test cases to the concrete operation test class e.g. `BalanceOperationTest`, `PurchaseOperationTest` etc.
3. `FruitShopServiceImplTest` should have simple test cases for testing processing of  transaction in the loop.